### ISSUE TITLE
Move device fails with 'arch' KeyError

### ISSUE THREAD

**Issue by jonathankehler** [balena-io/balena-sdk-python#240]
Exception when trying to move a device to an application - since 11.2.0.

The error occurs in `/usr/local/lib/python3.6/dist-packages/balena/models/device.py(951)`
```
if not self.device_os.is_architecture_compatible_with(device_dev_type['arch'], app_dev_type['arch']):
```

Because `device_dev_type` has no key 'arch'.

This happens in 11.2.0 and not in 11.1.0 (just confirmed).

In both version I reference the same device in Balena and trying to move to same application.

It happens with both arm and amd devices.

**Comment by jonathankehler** [comment]
```
(Pdb++) device_dev_type                                                                                                                                                                                     
{'id': 145, 'slug': 'genericx86-64-ext', 'name': 'Generic x86_64', 'is_private': False, 'is_of__cpu_architecture': {'__id': 2, '__deferred': {'uri': '/resin/cpu_architecture(@id)?@id=2'}}, 'belongs_to__de
vice_family': None, '__metadata': {'uri': '/resin/device_type(@id)?@id=145'}} 
```

**Comment by nghiant2710** [comment]
Hey @jonathankehler, thank you for the report. We've fixed this issue in [v11.3.1](https://github.com/balena-io/balena-sdk-python/commit/f9c8c8f86fe3a816609d9c5af1631bfd2ae9c994) please update your SDK and give it a try.
